corehq/apps/app_manager/models.py

```diff
@@ -4410,15 +4433,18 @@ def odk_profile_display_url(self):
     def odk_media_profile_display_url(self):
         return self.short_odk_media_url or self.odk_media_profile_url
 
-    def get_odk_qr_code(self, with_media=False):
+    def get_odk_qr_code(self, with_media=False, build_profile_id=None):
         """Returns a QR code, as a PNG to install on CC-ODK"""
         try:
             return self.lazy_fetch_attachment("qrcode.png")
         except ResourceNotFound:
             from pygooglechart import QRChart
             HEIGHT = WIDTH = 250
             code = QRChart(HEIGHT, WIDTH)
-            code.add_data(self.odk_profile_url if not with_media else self.odk_media_profile_url)
+            url = self.odk_profile_url if not with_media else self.odk_media_profile_url
+            if build_profile_id:
```

Comment:
small: overly permissive. consider changing to `if build_profile_id is None:`. If all other false values are assumed to be not valid here, then maybe `assert build_profile_id, repr(build_profile_id)` on the next line (it's likely a bug if `build_profile_id` is any value that evaluates to false other than `None`).